e107_handlers/cache_handler.php

```diff
@@ -146,13 +144,13 @@ public function retrieve($CacheTag, $MaximumAge = false, $ForcedCheck = false, $
 				else
 				{
 					$ret = file_get_contents($cache_file);
-					if (substr($ret,0,strlen(self::CACHE_PREFIX)) == self::CACHE_PREFIX)
+					if (strpos($ret,self::CACHE_PREFIX) === 0 )  
 					{
 						$ret = substr($ret, strlen(self::CACHE_PREFIX));
 					}
-					elseif(substr($ret,0,5) == '<?php')
+					elseif(strpos($ret,'<?php') === 0)
 					{
-						$ret = substr($ret, 5);		// Handle the history for now
+						$ret = '<?php' ;			// Handle the history for now
```

Comment:
Corrected... 
Render time: 0.515 cpu sec (40.00% load, 0.058 startup). Clock: 1.30 sec (6% of that for queries) DB queries: 213. Memory: 11.69MB/14.00MB

![image](https://cloud.githubusercontent.com/assets/2847651/22399714/d5ba6434-e59a-11e6-8070-ae6911a4a08e.png)
![image](https://cloud.githubusercontent.com/assets/2847651/22399722/09deae50-e59b-11e6-8189-c7c58b9d8406.png)